<!-- launch file for Yonah main SMS Telemetry Node on Bone as a standalone (for testing purposes)-->

<launch>

    <!-- please fill in the ground router's number. Enclose it in '' to recognise it as a string (instead of int)-->
    <arg name="GCS_no" default="'12345678'"/>

    <!-- phone numbers authorized to send commands to aircraft. Whitelist should be in same folder as nodes-->
    <arg name="whitelist" default="$(find sms)/src/whitelist.txt"/>

    <!--Various regular payload intervals in seconds-->
    <arg name="interval_1" default="10"/>
    <arg name="interval_2" default="60"/>

    <!-- screen = output to roslaunch screen for debug; log = log to rosout-->
    <arg name="log_output" default="screen"/>

    <!-- hostname and IP of onboard cellular router -->
    <arg name="router_hostname" default="root@192.168.1.1" />

    <!-- SMS_rx node: Receive SMS from Ground Control and send commands to aircraft/SMS_tx if necessary-->
    <node pkg="sms" type="SMS_rx.py" name="SMS_rx"
    respawn="true" respawn_delay="3" output="$(arg log_output)">
        <param name="whitelist" value="$(arg whitelist)" />
        <param name="router_hostname" value = "$(arg router_hostname)"/>
    </node>

    <!--SMS_tx node: Send SMS containing key aircraft info to Ground Control.
    SMS sending is deactivated by default. If SMS_rx receives a message by GCS requesting for SMS sending,
    then SMS_rx will notify SMS_tx to start sending (similar for deactivation)-->
    <node pkg="sms" type="SMS_tx.py" name="SMS_tx"
    respawn="true" respawn_delay="3" output="$(arg log_output)">
        <param name="GCS_no" value="$(arg GCS_no)" />
        <param name="router_hostname" value = "$(arg router_hostname)"/>
    </node>

        <!--air_despatcher node-->
    <node pkg="despatcher" type="air_despatcher.py" name="air_despatcher"
    respawn="true" respawn_delay="3" output="$(arg log_output)">
        <param name="interval_1" value="$(arg interval_1)" />
        <param name="interval_2" value="$(arg interval_2)" />
    </node>

</launch>
